internal/router/router.go

```diff
@@ -29,5 +30,10 @@ func public(app *fiber.App) {
 
 func restricted(app *fiber.App) {
 	app.Post("/v1/signup", controller.CompleteSignUp)
+	// TODO: move to agora after api gateway setup
 	app.Get("/v1/me", controller.GetMyAccountInfo)
+
+	gateway := app.Group("/gateway")
+	gateway.Use(middleware.NewGatewayCheck())
+	gateway.Get("/v1/token", controller.CheckToken)
```

Comment:
I think this code will work synchronously, can you explain it to me? (What is the purpose for grouping such endpoints?: `/gateway`)